lib/src/Base/Func/SymbolicParserExprTk.cxx

```diff
@@ -82,7 +82,7 @@ Point SymbolicParserExprTk::operator()(const Point & inP) const
     std::copy(inputStack_.begin() + inputDimension, inputStack_.end(), result.begin());
     for (UnsignedInteger outputIndex = 0; outputIndex < outputDimension; ++ outputIndex)
     {
-      if (!SpecFunc::IsNormal(result[outputIndex]))
+      if (checkResult_ && !SpecFunc::IsNormal(result[outputIndex]))
```

Comment:
The whole for-loop is useless if `checkResult_` is false, this block should be written instead
```
if (checkResult_)
  for...
```

BTW the previous loop should also be rewritten IMO
```
    if (checkResult_)
    {
      for (UnsignedInteger outputIndex = 0; outputIndex < result.getDimension(); ++ outputIndex)
      {
        const Scalar value = expressions_[outputIndex]->value();
        // ExprTk does not throw on domain/division errors
        if (!SpecFunc::IsNormal(value))
          throw InternalException(HERE) << "Cannot evaluate " << formulas_[outputIndex] << " at " << inputVariablesNames_.__str__() << "=" << inP.__str__();
        result[outputIndex] = value;
      }
    }
    else
    {
      for (UnsignedInteger outputIndex = 0; outputIndex < result.getDimension(); ++ outputIndex)
        result[outputIndex] = expressions_[outputIndex]->value();
    }
```

This applies of course to both parsers.

Reply:
@dbarbier good idea